### ISSUE TITLE
Failed to load PMD resources on multi-threaded maven build

### ISSUE THREAD

**Issue by longtimeago** [yegor256/qulice#933]
```java
Oct 05, 2018 11:10:03 AM net.sourceforge.pmd.RulesetsFactoryUtils getRuleSets
SEVERE: Ruleset not found
net.sourceforge.pmd.RuleSetNotFoundException: Can't find resource 'category/java/errorprone.xml' for rule 'AvoidDuplicateLiterals'.  Make sure the resource is a valid file or URL and is on the CLASSPATH. Here's the current classpath: /home/me/.asdf/installs/maven/3.5.0/boot/plexus-classworlds-2.5.2.jar
	at net.sourceforge.pmd.RuleSetReferenceId.getInputStream(RuleSetReferenceId.java:405)
	at net.sourceforge.pmd.RuleSetFactory.parseRuleSetNode(RuleSetFactory.java:330)
	at net.sourceforge.pmd.RuleSetFactory.createRuleSet(RuleSetFactory.java:214)
	at net.sourceforge.pmd.RuleSetFactory.createRule(RuleSetFactory.java:313)
	at net.sourceforge.pmd.RuleSetFactory.parseRuleReferenceNode(RuleSetFactory.java:609)
	at net.sourceforge.pmd.RuleSetFactory.parseRuleNode(RuleSetFactory.java:463)
	at net.sourceforge.pmd.RuleSetFactory.parseRuleSetNode(RuleSetFactory.java:367)
	at net.sourceforge.pmd.RuleSetFactory.createRuleSet(RuleSetFactory.java:214)
	at net.sourceforge.pmd.RuleSetFactory.createRuleSet(RuleSetFactory.java:209)
	at net.sourceforge.pmd.RuleSetFactory.createRuleSets(RuleSetFactory.java:170)
	at net.sourceforge.pmd.RuleSetFactory.createRuleSets(RuleSetFactory.java:154)
	at net.sourceforge.pmd.RulesetsFactoryUtils.getRuleSets(RulesetsFactoryUtils.java:38)
	at net.sourceforge.pmd.processor.AbstractPMDProcessor.createRuleSets(AbstractPMDProcessor.java:68)
	at net.sourceforge.pmd.processor.AbstractPMDProcessor.processFiles(AbstractPMDProcessor.java:101)
	at net.sourceforge.pmd.PMD.processFiles(PMD.java:329)
	at com.qulice.pmd.SourceValidator.validateOne(SourceValidator.java:136)
	at com.qulice.pmd.SourceValidator.validate(SourceValidator.java:102)
	at com.qulice.pmd.PmdValidator.validate(PmdValidator.java:70)
	at com.qulice.maven.CheckMojo$ValidatorCallable.call(CheckMojo.java:235)
	at com.qulice.maven.CheckMojo$ValidatorCallable.call(CheckMojo.java:203)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

Sometimes it fails with on `Can't find resource 'category/java/documentation.xml' for rule 'null'`

qulice version 0.18.2
```
$ mvn -v
Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T22:39:06+03:00)
Maven home: /home/me/.asdf/installs/maven/3.5.0
Java version: 1.8.0_151, vendor: Oracle Corporation
Java home: /home/m/.asdf/installs/java/8.151/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-36-generic", arch: "amd64", family: "unix"
```

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) please, pay attention to this issue

**Comment by 0crat** [comment]
@longtimeago[/z](https://www.0crat.com/u/longtimeago) this project will fix the problem faster if you donate a few dollars to it; just [click here](https://www.0crat.com/contrib/C3T49A35L) and pay via Stripe, it's very fast, convenient and appreciated; thanks a lot!

**Comment by longtimeago** [comment]
upgrade to PMD 6.8.0 could help

**Comment by longtimeago** [comment]
nah ... tried locally to upgrade PMD to 6.8.0 - the issue is still there :(

**Comment by krzyk** [comment]
@longtimeago does it fail always? I tried it on other projects and haven't seen such error.
Maybe there is some specific code that causes it?

**Comment by longtimeago** [comment]
@krzyk not always but 2 out of 3.
We have >20 top-level maven modules with 1-4 sub-modules per each.

And running maven with following params
mvn -T 8 -B -Ddependency.locations.enabled=false verify

**Comment by longtimeago** [comment]
it works fine with qulice 0.17.2

**Comment by krzyk** [comment]
@longtimeago Now I noticed it, multithreaded maven build. OK, that I didn't test.

**Comment by krzyk** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/teamed/qulice/issues/933#issuecomment-427302963))

@krzyk Job [#933](https://github.com/teamed/qulice/issues/933) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/teamed/qulice/issues/933#issuecomment-427302963))

@krzyk Thanks for your contribution, @longtimeago[/z](https://www.0crat.com/u/longtimeago)! If you would be a member of [the project](https://www.0crat.com/p/C3T49A35L), you would now earn +15 reputation points, as explained in [§29](http://www.zerocracy.com/policy.html#29). You can join and apply to it, see [§2](http://www.zerocracy.com/policy.html#2).

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by paulodamaso** [comment]
@krzyk Looks like it has already been resolved [here](https://github.com/teamed/qulice/issues/933#issuecomment-427299920)

**Comment by krzyk** [comment]
@paulodamaso It is a regression, in 0.17.2 it worked but the bug report is for 0.18.2

**Comment by paulodamaso** [comment]
@krzyk Could not reproduce the problem

**Comment by paulodamaso** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/teamed/qulice/issues/933#issuecomment-457966869))

@paulodamaso The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#933](https://github.com/teamed/qulice/issues/933), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
@krzyk[/z](https://www.0crat.com/u/krzyk) everybody who has role `DEV` is banned at [#933](https://github.com/teamed/qulice/issues/933); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by yegor256** [comment]
in the latest version it's fixed (0.23.0)
